### ISSUE TITLE
Better DL1CameraContainer organization

### ISSUE THREAD

**Issue by kosack** [cta-observatory/ctapipe#1849]
## Problem

The axiom that `Container`s should contain data items that is filled all at once (logically grouped by when they are created) is a bit broken for `DL1CameraContainer`.    In #1810, extractors now generate a DL1CameraContainer, with the images and validity flag.  However they currently contain in addition:
* the cleaned image mask
* the parameters (ImageParametersContainer) derived from the mask applied to the images

which get filled later.

## Suggested solution 
It would be  better to have something like the following that separates these two things that are not filled simultaneously:
```py3
class DL1ImageContainer:   # (formerly DL1CameraContainer)
    image = ...
    peak_time = ...
    is_valid = ...

class DL1ParametersContainer:  # could even support more than one of these
    image_mask = ...
    parameters = ImageParametersContainer  # parameters that used the image_mask
```
These are basically "DL1-IMAGE" (DL1a) and "DL1-PARAMETERS" (DL1b), which are stored separately in any case.

That would be nice, as then *Image*Extractors produce DL1*Image*Containers.

_Originally posted by @kosack in https://github.com/cta-observatory/ctapipe/pull/1810#issuecomment-1068194067_

**Comment by kosack** [comment]
I guess then `DL1CameraContainer` (or whatever we want to call it, maybe `DL1TelescopeContainer` would be closer to the data model naming scheme) would contain both of these classes:

```py3
class DL1TelescopeContainer:
   image = DL1ImageContainer
   parameters = DL1ParametersContainer
```
Howver, this is not so nice, as you then get things like `event.dl1.tel[i].image.image` and `event.dl1.tel[i].parameters.parameters`.   Any ideas for a better name? It could also just be "a", and "b"... (`event.dl1.tel[x].a.image`), but that's also not so readable.

Alternately, we change DL1ImageContainer's "image" field to be "intensity", and then you have: `event.dl1.tel[i].image.intensity` and `event.dl1.tel[i].image.peak_time` (and any other images we eventually add like peak_width, etc)

**Comment by maxnoe** [comment]
> event.dl1.tel[i].image.image

Image is not really clear, so maybe `event.dl1.tel[i].image.photoelectrons` / `event.dl1.tel[i].image.peak_time` would be clearer and avoid the duplication?

> event.dl1.tel[i].parameters.parameters 

I don't see why this is necessary. The `DL1ParametersContainer` should just be the current `ParametersContainer`, not:

```
class DL1ParametersContainer:  # could even support more than one of these
    image_mask = ...
    parameters = ImageParametersContainer  # parameters that used the image_mask
```

but:
```
class DL1ParametersContainer:  # could even support more than one of these
    image_mask = ...
    hillas = HillasParametersContainer  
    morphology = ...
   ...
```

The `image_mask` is something in between the parameters and the image. I think it's possible to argue for each attaching it to the image container (adding a default of None is also not expensive), where it would fit better with regard to the shape,
attaching it to the parameters or attaching it to the base container.

**Comment by maxnoe** [comment]
I think I would favor having the mask still in the image, essentially just adding a parent `DL1TelescopeContainer`

```
DL1TelescopeContainer
    image
        photo_electrons
        peak_time
        image_mask
    parameters
        hillas
        morphology
        timing
        ....
```

**Comment by maxnoe** [comment]
Returning a `DL1ImageContainer` from the extractors would also have the nice effect then of overwriting the image mask, making it impossible to have a mask from an older image.

**Comment by kosack** [comment]
> I think I would favor having the mask still in the image part, essentially just adding a parent DL1TelescopeContainer

I originally liked it that way as well (as it is an "image"), but there are two problems"
1. `image_mask` is not filled when the other items in the container are filled (Extractor generates the intensitty and peak time images, but not the cleaning mask).  This means that Extractors create an object that is incomplete.
2. The cleaning mask and the parameters are logically connected: the parameters depend on the choice of cleaning. In principle you could have multiple cleanings and multiple sets of parameters associated with them.

**Comment by kosack** [comment]
>         photo_electrons

I don't think we should name it by its units, as in principle we aren't required to have this image in PE units. It's an image of Cherenkov light *intensity* or *brightness*.  Which units we use depend on how we calibrate it, which could change in the future.

**Comment by maxnoe** [comment]
> I don't think we should name it by its units

Now we are getting into difficult terrain... for me `photo_electrons` is a unitless quantity, not a unit.

**Comment by maxnoe** [comment]
> : the parameters depend on the choice of cleaning

And the mask depends on the image... As said, it sits squarely in between both
